#! /bin/sh /usr/share/dpatch/dpatch-run
## 10_fix_clean_target.dpatch by Henning Glawe <glaweh@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Fix clean target in Makefile.PLs

@DPATCH@
diff -urNad pdl-2.4.2+cvs.20050501/Basic/Core/Makefile.PL /tmp/dpep.0Odhl6/pdl-2.4.2+cvs.20050501/Basic/Core/Makefile.PL
--- pdl-2.4.2+cvs.20050501/Basic/Core/Makefile.PL	2005-05-03 09:18:00.562118240 +0200
+++ /tmp/dpep.0Odhl6/pdl-2.4.2+cvs.20050501/Basic/Core/Makefile.PL	2005-05-03 09:27:29.213670064 +0200
@@ -110,7 +110,7 @@
  'clean'        => {'FILES'  => 'pdlcore$(OBJ_EXT) pdlapi$(OBJ_EXT) '.
                    'pdlhash$(OBJ_EXT) pdlconv$(OBJ_EXT) pdlsections$(OBJ_EXT) '.
                    'pdlconv.c pdlsections.c pdlcore.c '.
-		   'pdl.h pdlsimple.h pdlcore.h '.
+		   'pdl.h pdlsimple.h pdlcore.h Config.pm '.
 		   'Types.pm Version.pm Core.pm Core.c Core.xs ' .
 		   'mymalloc.p badsupport.p'
 		   },
diff -urNad pdl-2.4.2+cvs.20050501/Basic/Gen/Makefile.PL /tmp/dpep.0Odhl6/pdl-2.4.2+cvs.20050501/Basic/Gen/Makefile.PL
--- pdl-2.4.2+cvs.20050501/Basic/Gen/Makefile.PL	2005-05-02 09:42:28.000000000 +0200
+++ /tmp/dpep.0Odhl6/pdl-2.4.2+cvs.20050501/Basic/Gen/Makefile.PL	2005-05-03 09:26:31.785400488 +0200
@@ -13,7 +13,7 @@
 WriteMakefile(NAME => "PDL::PP",
 	PM => {@pms},
 	'EXE_FILES' => ['pptemplate'],
-	clean => {FILES => "PP/Dump.pm PP/Dump.pm.tmp pptemplate"},
+	clean => {FILES => "PP/Dump.pm PP/Dump.pm.tmp pptemplate pptemplate.pod"},
 );
 
 sub MY::postamble {
diff -urNad pdl-2.4.2+cvs.20050501/Basic/Pod/Makefile.PL /tmp/dpep.0Odhl6/pdl-2.4.2+cvs.20050501/Basic/Pod/Makefile.PL
--- pdl-2.4.2+cvs.20050501/Basic/Pod/Makefile.PL	2005-05-02 09:42:28.000000000 +0200
+++ /tmp/dpep.0Odhl6/pdl-2.4.2+cvs.20050501/Basic/Pod/Makefile.PL	2005-05-03 09:26:31.786400336 +0200
@@ -28,7 +28,7 @@
  'MAN1PODS' =>  { @man1 },
  'MAN3PODS' =>  { },
  'PM'    =>     { @pms },
- 'clean' => {FILES => ""},
+ 'clean' => {FILES => "PP-Inline.pod"},
  'linkext' =>   {LINKTYPE => ''},  # No linking required in this directory
 );
 
diff -urNad pdl-2.4.2+cvs.20050501/Demos/Makefile.PL /tmp/dpep.0Odhl6/pdl-2.4.2+cvs.20050501/Demos/Makefile.PL
--- pdl-2.4.2+cvs.20050501/Demos/Makefile.PL	2005-05-02 09:42:28.000000000 +0200
+++ /tmp/dpep.0Odhl6/pdl-2.4.2+cvs.20050501/Demos/Makefile.PL	2005-05-03 09:26:31.788400032 +0200
@@ -25,7 +25,8 @@
                     'Transform_demo.pm' => '$(INST_LIBDIR)/Demos/Transform_demo.pm',
                     'Cartography_demo.pm' => '$(INST_LIBDIR)/Demos/Cartography_demo.pm',
                     '../m51.fits'       => '$(INST_LIBDIR)/Demos/m51.fits',
-		   }
+		   },
+     'clean' => { 'FILES' => 'BAD_demo.pm BAD2_demo.pm' }
 );
 
 # For the lazy people
diff -urNad pdl-2.4.2+cvs.20050501/Graphics/TriD/OpenGL/Makefile.PL /tmp/dpep.0Odhl6/pdl-2.4.2+cvs.20050501/Graphics/TriD/OpenGL/Makefile.PL
--- pdl-2.4.2+cvs.20050501/Graphics/TriD/OpenGL/Makefile.PL	2005-05-02 09:42:28.000000000 +0200
+++ /tmp/dpep.0Odhl6/pdl-2.4.2+cvs.20050501/Graphics/TriD/OpenGL/Makefile.PL	2005-05-03 09:26:31.789399880 +0200
@@ -11,7 +11,7 @@
 push @{$hash{LIBS}}, $PDL::Config{OPENGL_LIBS};
 $hash{INC} .= $PDL::Config{OPENGL_INC};
 $hash{DEFINE} .= $PDL::Config{OPENGL_DEFINE};
-$hash{clean}{FILES} .= ' tmp_*.h ppcode.out';
+$hash{clean}{FILES} .= ' *.h.cpp tmp_*.h ppcode.out typemap';
 
 #use Data::Dumper;
 #my $out = Dumper(\%hash);
diff -urNad pdl-2.4.2+cvs.20050501/Lib/Slatec/Makefile.PL /tmp/dpep.0Odhl6/pdl-2.4.2+cvs.20050501/Lib/Slatec/Makefile.PL
--- pdl-2.4.2+cvs.20050501/Lib/Slatec/Makefile.PL	2005-05-03 09:18:21.413948280 +0200
+++ /tmp/dpep.0Odhl6/pdl-2.4.2+cvs.20050501/Lib/Slatec/Makefile.PL	2005-05-03 09:26:31.791399576 +0200
@@ -101,7 +101,7 @@
 }
 
 $hash{LIBS}[0] .= $f77->runtime ;
-$hash{clean}{FILES} .= " f77_underscore" .
+$hash{clean}{FILES} .= " SlatecProtos.h f77_underscore" .
   join '', map {" slatec/$_.o "} @slatecfiles;
 
 # Handle multiple compilers
